examples/demo4.py

```diff
@@ -133,16 +150,32 @@ def main(argv):
         # Take the (e1, e2) shape parameters from the catalog as well.
         psf = psf.shear(e1=cat.getFloat(k,2), e2=cat.getFloat(k,3))
 
-        # Galaxy is a bulge + disk with parameters taken from the catalog:
-        disk = galsim.Exponential(flux=0.6, half_light_radius=cat.getFloat(k,5))
-        disk = disk.shear(e1=cat.getFloat(k,6), e2=cat.getFloat(k,7))
+        # Galaxy is a bulge + disk(+knots) with parameters taken from the catalog:
+
+        # put some fraction of the disk light into knots of star formation
 
-        bulge = galsim.DeVaucouleurs(flux=0.4, half_light_radius=cat.getFloat(k,8))
-        bulge = bulge.shear(e1=cat.getFloat(k,9), e2=cat.getFloat(k,10))
+        disk_hlr = cat.getFloat(k,5)
+        disk_e1 = cat.getFloat(k,6)
+        disk_e2 = cat.getFloat(k,7)
+        bulge_hlr = cat.getFloat(k,8)
+        bulge_e1 = cat.getFloat(k,9)
+        bulge_e2 = cat.getFloat(k,10)
+
+        smooth_disk = galsim.Exponential(flux=smooth_disk_frac, half_light_radius=disk_hlr)
+
+        knots = galsim.RandomWalk(n_knots, disk_hlr, flux=knot_frac)
```

Comment:
I agree with your statement about needing 100 for an irregular, but in this case there is a bulge+disk component so why not go with 10-20, and have a comment explicitly noting that that's only sufficient because of the existence of those components?  That way you've given that information, but the demo is still nearly as fast as before.